doc/concepts/wai/README.md

```diff
@@ -0,0 +1,110 @@
+# WebAssembly Interfaces (WAI)
+
+The WebAssembly spec that was first released in 2017 was only a minimum viable
+product and deliberately left several features incomplete to be iterated on by
+the ecosystem.
+
+Arguably the most important functionality gap is the fact that only WebAssembly
+primitives can be passed between the host and guest. That means imports and
+exports can only use the following data types,
+
+- `i32` - signed 32-bit integers
+- `i64` - signed 64-bit integers
+- `f32` - a 32-bit float
+- `f64` - a 64-bit float (often called a `double`)
+- `funcref` - a reference to a WebAssembly function
+- `externref` - a reference to some opaque object owned by the WebAssembly
+  virtual machine
+
+You'll notice this list doesn't even include strings or boolean values!
+
+The WebAssembly Interfaces project (WAI for short) provides a polyfill for
+passing around higher-level objects. It lets developers define their imports and
+exports in a `*.wai` file, then uses `wai-bindgen` to generate glue which
+automagically passes things around within the constraints of WebAssembly.
+
+There are four main parts to WAI:
+
+- The `*.wai` file
+- the WAI Bindgen code generator
+- The guest
+- The host
+
+The [Wasmer Pack](../../) project provides a convenient way to consume
+WebAssembly packages which implement a WAI interface.
+
+Some useful links:
+
+- [The `wasmerio/wai` repository](https://github.com/wasmerio/wai)
+- [The `*.wai` format](https://github.com/wasmerio/wai/blob/main/WAI.md)
+- [The `wai-bindgen` CLI on crates.io](https://crates.io/crates/wai-bindgen-cli)
+
+## The `*.wai` File
+
+WAI uses a file (or files) with the `*.wai` extension to define the host-guest
+interface for an application that uses WebAssembly.
+
+The items in a `*.wai` file map closely to concepts shared by most programming
+languages. It has [types](./types.md), interfaces
+(["Resources"](./resources.md)), structs (["Records"](./records.md)),
+[functions](./functions.md), [enums](./variants.md), and so on.
+
+The precise syntax [is defined in the WAI repository][wai-format] and a parser,
+[`wai-parser`][wai-parser], is available as a Rust crate.
+
+## WAI Bindgen
+
+The WAI Bindgen code generator consumes `*.wai` files and generates glue code
+that [the host](#the-host) can use for using functionality from a WebAssembly
+module or [the guest](#the-guest) can use for implementing that functionality.
+
+There are two primary ways users will interact with WAI Bindgen, the
+`wai-bindgen` CLI, and the `wai-bindgen-*` family of crates.
+
+The `wai-bindgen` CLI provides a command-line interface to the `wai-bindgen-*`
+crates, and is often used for once-off investigation or integration with a
+non-Rust build system.
```

Comment:
Does **non-Rust** have correct capitalization ?

Reply:
Yep. Rust is a noun, so it will be capitalised even if it has a `non-` prefix.